File: edwincase/src/main/java/com/edwin/edwincase/EdwinApplication.java
package com.edwin.edwincase;

import android.app.Application;
import android.content.Context;

/**
 * Created by Edwin on 2017/12/17.
 */

public class EdwinApplication extends Application {

    private static Context mContext;
    private static Application sApplication;
    @Override
    public void onCreate() {
        super.onCreate();
        mContext = getApplicationContext();
        sApplication = new EdwinApplication();
    }

    public static Application getInstance() {
        return sApplication;
    }

    public static Context getContext() {
        return mContext;
    }
}

File: app/src/main/java/com/edwin/utils/EdwinApplication.kt
package com.edwin.utils

import android.app.Application
import android.content.Context

/**
 * Created by hongy_000 on 2017/10/19.
 */
class EdwinApplication : Application() {


    override fun onCreate() {
        super.onCreate()
        edwinApplication = this
    }

    companion object {
        private var edwinApplication :EdwinApplication? = null
        fun getContext(): Context {
            return edwinApplication as Context
        }
    }
}